### ISSUE TITLE
Error running OrthoMCL on the BLAST output

### ISSUE THREAD

**Issue by mudithekanayake** [rhysf/Synima#32]
Hello,
I am trying to do a synteny analysis and I have run first two steps in the pipeline successfully. I think they ran successfully since they did not fail. However when I try to run the OrthoMCL step I get the following error. Can you please help me to solve this?

These are the first two commands I ran.
`perl ../util/Create_full_repo_sequence_databases.pl -r ./Repo_spec.txt -f gene`
`perl ../util/Blast_grid_all_vs_all.pl -r ./Repo_spec.txt`

This is the command which failed.
`perl ../util/Blast_all_vs_all_repo_to_OrthoMCL.pl -r ./Repo_spec.txt`

Tnis is the error I got.
> retrieve_blast_pairs: Printing to OMCL_outdir/all_blast_pairs.m8
> Saving ID->Genome from ./Repo_spec.txt.all.GFF3...
> Assigning genome codes...
> Ndig => G001
> Emu => G002
> Writing Gcoded version of blast.m8...
> Error, no genome for g33007.t1 saved from all_annotations.gff3 (check settings and rerun): Inappropriate ioctl for device

If needed I am posting the logs from the first two commands as well. Please let me know if there are any issues.
`perl ../util/Create_full_repo_sequence_databases.pl -r ./Repo_spec.txt -f gene`
> Indexing Emu...
> Indexing Ndig...
> Creating repository sequence databases...
> Copying ./Emu/Emu_final.annotation.pep to ./Repo_spec.txt.all.PEP...
> Copying ./Emu/Emu_final.annotation.cds to ./Repo_spec.txt.all.CDS...
> Copying ./Ndig/Ndig_braker.annotation.pep to ./Repo_spec.txt.all.PEP...
> Copying ./Ndig/Ndig_braker.annotation.cds to ./Repo_spec.txt.all.CDS...
> Combine_all_gff3_files_in_repo: printing to ./Repo_spec.txt.all.GFF3
> trying to remove ID=
> Ndig: Saving IDs (e.g. jg16728). If fail, check settings for matching FASTA and GFFs...
> Found 23497 gene features in ./Ndig/Ndig_braker.annotation.gff3
> trying to remove ID=
> Emu: Saving IDs (e.g. Em0001g1a). If fail, check settings for matching FASTA and GFFs...
> Found 39246 gene features in ./Emu/Emu_final.annotation.gff3
> Checking FASTA and GFF repository sequence databases match...
> fasta_id_to_seq_hash: saving sequences from ./Repo_spec.txt.all.PEP...
> gff_to_contig_parent_to_cds_hash: saving all from ./Repo_spec.txt.all.GFF3...
> 
> 
> 
> 39245 / 62741 found
> ./Repo_spec.txt.all.PEP and ./Repo_spec.txt.all.GFF3 repository sequence databases are not correctly formatted. Change settings and re-run, or rename ID's in FASTA or GFF to match.
> fasta_id_to_seq_hash: saving sequences from ./Repo_spec.txt.all.CDS...
> gff_to_contig_parent_to_cds_hash: saving all from ./Repo_spec.txt.all.GFF3...
> 
> 
> 
> 0 / 62741 found
> ./Repo_spec.txt.all.CDS and ./Repo_spec.txt.all.GFF3 repository sequence databases are not correctly formatted. Change settings and re-run, or rename ID's in FASTA or GFF to match.
> Finished.

`perl ../util/Blast_grid_all_vs_all.pl -r ./Repo_spec.txt`

> fasta_to_struct: saving from ./Repo_spec.txt.all.PEP...
> Splitting seq dictionary for blasting...
> Emu = ./Emu/Emu_final.annotation.pep.synima-parsed.PEP...
> Ndig = ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP...
> Using legacy BLAST...
> CMD: formatdb -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -p T
> CMD: formatdb -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -p T
> blastall -p blastp -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -d ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -m 8 -v 1000 -b 1000 -e 1e-20 > ./Ndig/RBH_blast_PEP/vs_Ndig.blast.m8
> [blastall] WARNING: jg155.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -d ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -m 8 -v 5 -b 5 -e 1e-20 > ./Ndig/RBH_blast_PEP/vs_Emu.blast.m8
> [blastall] WARNING: jg155.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -d ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -m 8 -v 5 -b 5 -e 1e-20 > ./Emu/RBH_blast_PEP/vs_Ndig.blast.m8
> [blastall] WARNING: Em0003g1562a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0008g20a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0202g8a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0964g2a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0660g2a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -d ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -m 8 -v 1000 -b 1000 -e 1e-20 > ./Emu/RBH_blast_PEP/vs_Emu.blast.m8
> [blastall] WARNING: Em0003g1562a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0008g20a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0202g8a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0964g2a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0660g2a: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options

**Comment by rhysf** [comment]
Hi @mudithekanayake 

You need to redo the Create_full_repo_sequence_databases step, after identifying the difference between your ids in your GFF and your FASTA file. If you look at the output:

**39245 / 62741 found**
./Repo_spec.txt.all.PEP and ./Repo_spec.txt.all.GFF3 repository sequence databases are not correctly formatted. Change settings and re-run, or rename ID's in FASTA or GFF to match.
fasta_id_to_seq_hash: saving sequences from ./Repo_spec.txt.all.CDS...
gff_to_contig_parent_to_cds_hash: saving all from ./Repo_spec.txt.all.GFF3...

**0 / 62741 found**
./Repo_spec.txt.all.CDS and ./Repo_spec.txt.all.GFF3 repository sequence databases are not correctly formatted. Change settings and re-run, or rename ID's in FASTA or GFF to match.
Finished.

This shows that 39,245 / 62,741 peptide sequences have been correctly matched between the GFF and FASTA (which are very likely your Emu sequences), and 0 (none) of your DNA/CDA have been matched. 

Therefore you should re-look at both of your CDS files (which should have identical ids to your PEP files), and figure out why your Ndig id's are not matching between your GFF and either PEP or CDS.

**Comment by mudithekanayake** [comment]
Hello @rhysf 
Thank you for your help. I have figured out the differences between the sequence IDs in the files and my current logs from the first two steps look like this.

`perl ../util/Create_full_repo_sequence_databases.pl -r ./Repo_spec.txt`
> Indexing Emu...
> Indexing Ndig...
> Creating repository sequence databases...
> Copying ./Emu/Emu_final.annotation.pep to ./Repo_spec.txt.all.PEP...
> Copying ./Emu/Emu_final.annotation.cds to ./Repo_spec.txt.all.CDS...
> Copying ./Ndig/Ndig_braker.annotation.pep to ./Repo_spec.txt.all.PEP...
> Copying ./Ndig/Ndig_braker.annotation.cds to ./Repo_spec.txt.all.CDS...
> Combine_all_gff3_files_in_repo: printing to ./Repo_spec.txt.all.GFF3
> trying to remove ID=
> Ndig: Saving IDs (e.g. jg7544.t1). If fail, check settings for matching FASTA and GFFs...
> Found 24569 mRNA features in ./Ndig/Ndig_braker.annotation.gff3
> trying to remove ID=
> Emu: Saving IDs (e.g. Em0001g1a.t1). If fail, check settings for matching FASTA and GFFs...
> Found 39246 mRNA features in ./Emu/Emu_final.annotation.gff3
> Checking FASTA and GFF repository sequence databases match...
> fasta_id_to_seq_hash: saving sequences from ./Repo_spec.txt.all.PEP...
> gff_to_contig_parent_to_cds_hash: saving all from ./Repo_spec.txt.all.GFF3...
> 
> 
> 
> 63813 / 63813 found
> ./Repo_spec.txt.all.PEP and ./Repo_spec.txt.all.GFF3 repository sequence databases are correctly formatted.
> fasta_id_to_seq_hash: saving sequences from ./Repo_spec.txt.all.CDS...
> gff_to_contig_parent_to_cds_hash: saving all from ./Repo_spec.txt.all.GFF3...
> 
> 
> 
> 63813 / 63813 found
> ./Repo_spec.txt.all.CDS and ./Repo_spec.txt.all.GFF3 repository sequence databases are correctly formatted.
> Finished.


`perl ../util/Blast_grid_all_vs_all.pl -r ./Repo_spec.txt`
> fasta_to_struct: saving from ./Repo_spec.txt.all.PEP...
> Splitting seq dictionary for blasting...
> Emu = ./Emu/Emu_final.annotation.pep.synima-parsed.PEP...
> Ndig = ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP...
> Using legacy BLAST...
> CMD: formatdb -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -p T
> CMD: formatdb -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -p T
> blastall -p blastp -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -d ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -m 8 -v 1000 -b 1000 -e 1e-20 > ./Ndig/RBH_blast_PEP/vs_Ndig.blast.m8
> [blastall] WARNING: jg155.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -d ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -m 8 -v 5 -b 5 -e 1e-20 > ./Ndig/RBH_blast_PEP/vs_Emu.blast.m8
> [blastall] WARNING: jg155.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -d ./Ndig/Ndig_braker.annotation.pep.synima-parsed.PEP -m 8 -v 5 -b 5 -e 1e-20 > ./Emu/RBH_blast_PEP/vs_Ndig.blast.m8
> [blastall] WARNING: Em0003g1562a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0964g2a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0202g8a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0660g2a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0008g20a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> blastall -p blastp -i ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -d ./Emu/Emu_final.annotation.pep.synima-parsed.PEP -m 8 -v 1000 -b 1000 -e 1e-20 > ./Emu/RBH_blast_PEP/vs_Emu.blast.m8
> [blastall] WARNING: Em0003g1562a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0964g2a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0202g8a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0660g2a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options
> [blastall] WARNING: Em0008g20a.t1: Could not calculate ungapped Karlin-Altschul parameters due to an invalid query sequence or its translation. Please verify the query sequence(s) and/or filtering options

But still I get this error when I run the third step.

`perl ../util/Blast_all_vs_all_repo_to_OrthoMCL.pl -r ./Repo_spec.txt`
> retrieve_blast_pairs: Printing to OMCL_outdir/all_blast_pairs.m8
> Saving ID->Genome from ./Repo_spec.txt.all.GFF3...
> Assigning genome codes...
> Emu => G001
> Ndig => G002
> Writing Gcoded version of blast.m8...
> Error, no genome for g536.t1 saved from all_annotations.gff3 (check settings and rerun): Inappropriate ioctl for device

g536.t1 is present in all the three Ndig sequence files (cds, pep, gff3) and the genome annotation for this species is done by myself, so all the files were generated by me. Earlier issue was I haven't converted the gtf file to gff3 file properly and in Emu data there was not ".t1"s in the sequence ids of protein fasta file. After fixing those two errors, first step ran nicely, but again failed the third step giving the above error message.
I would be grateful to get your expertise insight in this issue.
Thank you.

**Comment by rhysf** [comment]
Hi, so i agree it looks like the first 2 steps worked, and the 3rd has not. The error appears to be that the script is unable to find which 'genome' g536.t1 belongs to/encodes it. The script is finding the gene with id g536.t1 in your blast m8 file, and looking for the genome ID from your 'synima parsed GFF' file (repo.all.GFF3 = 2nd column).

If you grep for 'g536.t1' - what is the line in that file? and does that genome correspond with the folder name?

**Comment by mudithekanayake** [comment]
Hi @rhysf 
Thank you for the reply. Yes, the genome correspond with the folder name.

> [mudith@nova Ndig]$ ls
> Ndig_braker.annotation.cds
> Ndig_braker.annotation.gff3
> Ndig_braker.annotation.pep
> Ndig_braker.annotation.pep.synima-parsed.PEP
> Ndig_braker.annotation.pep.synima-parsed.PEP.phr
> Ndig_braker.annotation.pep.synima-parsed.PEP.pin
> Ndig_braker.annotation.pep.synima-parsed.PEP.psq
> Ndig.genome.fa
> RBH_blast_PEP

This is the Repo_spec.txt file.

```
//
Genome Emu
Annotation Emu_final
//
Genome Ndig
Annotation Ndig_braker
//
```
`grep 'g536.t1' Ndig_braker.annotation.gff3` gives this.

> contig_3519_np1212	AUGUSTUS	mRNA	39886	41449	.	+	.	ID=jg536.t1;Parent=jg536;
> contig_3519_np1212	AUGUSTUS	start_codon	39886	39888	.	+	0	ID=jg536.t1.start1;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	CDS	39886	40118	1	+	0	ID=jg536.t1.CDS1;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	exon	39886	40118	.	+	.	ID=jg536.t1.exon1;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	intron	40119	40164	.	+	.	ID=jg536.t1.intron1;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	CDS	40165	40675	0.74	+	1	ID=jg536.t1.CDS2;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	exon	40165	40675	.	+	.	ID=jg536.t1.exon2;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	intron	40676	40969	.	+	.	ID=jg536.t1.intron2;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	CDS	40970	41449	0.55	+	0	ID=jg536.t1.CDS3;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	exon	40970	41449	.	+	.	ID=jg536.t1.exon3;Parent=jg536.t1;
> contig_3519_np1212	AUGUSTUS	stop_codon	41447	41449	.	+	0	ID=jg536.t1.stop1;Parent=jg536.t1;
> contig_449_np1212	AUGUSTUS	stop_codon	1392202	1392204	.	-	0	ID=g536.t1.stop1;Parent=g536.t1;
> contig_449_np1212	AUGUSTUS	CDS	1392202	1395414	1	-	0	ID=g536.t1.CDS1;Parent=g536.t1;
> contig_449_np1212	AUGUSTUS	exon	1392202	1395414	.	-	.	ID=g536.t1.exon1;Parent=g536.t1;
> contig_449_np1212	AUGUSTUS	single	1392202	1395414	1	-	0	Parent=g536.t1;
> contig_449_np1212	AUGUSTUS	start_codon	1395412	1395414	.	-	0	ID=g536.t1.start1;Parent=g536.t1;

But `grep 'g536.t1' Repo_spec.txt.all.GFF3` gives this.

`contig_3519_np1212	Ndig	mRNA	39886	41449	.	+	.	jg536.t1`

jg536.t1 is a different sequence ID. So I think there is no g536.t1 in Repo_spec.txt.all.GFF3 file. But it is present in every other place including m8 files.

`grep 'g536.t1' vs_Ndig.blast.m8`

> jg18536.t1	g536.t1	52.71	996	397	9	29	959	81	1067	0.0	 979
> jg14203.t1	g536.t1	59.26	842	341	2	11	851	224	1064	0.0	 996
> g536.t1	g536.t1	100.00	1070	0	0	1	1070	1	1070	0.0	2017
> g536.t1	jg17937.t1	93.29	656	22	4	1	653	1	637	0.0	1106
> g536.t1	jg14203.t1	59.26	842	341	2	224	1064	11	851	0.0	 975
> g536.t1	jg18536.t1	52.71	996	397	9	81	1067	29	959	0.0	 944
> g536.t1	jg8990.t1	73.14	417	112	0	651	1067	14	430	0.0	 607
> g536.t1	jg1771.t1	99.61	255	1	0	816	1070	14	268	2e-160	 474
> jg8990.t1	g536.t1	72.79	430	117	0	1	430	638	1067	0.0	 644
> jg1771.t1	g536.t1	99.61	255	1	0	14	268	816	1070	7e-161	 474
> jg536.t1	jg536.t1	100.00	354	0	0	1	354	1	354	0.0	 748
> jg17937.t1	g536.t1	93.57	653	26	2	1	637	1	653	0.0	1105

`grep 'g536.t1' vs_Emu.blast.m8`

> g536.t1	Em0007g1054a.t1	31.77	299	184	7	166	447	823	1118	1e-34	 145
> g536.t1	Em0007g1054a.t1	30.03	303	184	7	166	447	2380	2675	2e-30	 131
> g536.t1	Em0007g1054a.t1	32.14	224	145	6	181	401	3822	4041	2e-23	 108
> g536.t1	Em0007g1043a.t1	35.89	248	154	4	202	447	2759	3003	3e-34	 144
> g536.t1	Em0007g1043a.t1	33.71	267	167	5	179	445	5606	5862	3e-32	 137
> g536.t1	Em0007g1043a.t1	29.08	282	192	5	167	447	1090	1364	4e-29	 127
> g536.t1	Em0007g1043a.t1	36.44	236	142	6	200	434	4169	4397	4e-29	 127
> g536.t1	Em0054g5a.t1	35.08	248	156	4	202	447	264	508	1e-33	 140
> g536.t1	Em0007g1126a.t1	29.64	334	212	7	132	447	28	356	1e-31	 134
> g536.t1	Em0007g1058a.t1	33.75	240	156	3	208	447	784	1020	2e-31	 134

Following are the g536.t1 sequence in cds file and pep file.

```
GTGTGGTCACTGGGTGTTCATGATAATCATTACGGTAACATCTCGTGTATATCACTCACG
TATGATGACCAGTATTTATTGACTGGTGCTGGTGATGGGAATGTTTTTATATATAAAACT
AATTTACCAGCAAACAGTCCAACCTCAGCTCCTGTCAAGAGTGCTGCTGGTGATAAAGTA
TTGAGAAGGGTTGATACTGGTGATGATATTGAGGATCCTTCACACTACAGTATTGAAGAT
GCTAAACAGAAAGCTGAACATGACAGGAAACTGAAAGCAGCTGAAGCCAAGAAACAGAAG
ctaacctgtgctgtgGGAATGGCCACGCGAGTGTGGAATCAGTTGCCTGGTGCTGCCTAT
TGGGGCACACTCCCTCCTCAAGTCTCAACAACTTCAGTTGGTGTGCAACCTTTAGAAAGG
AAACGTGCATTCTCTTCACCTACTTCACCAGTCAATGCAGTATAA
>jg2229.t1
atggaggttgagaaatcctttgggttgacattgagctttgagaagatcaagttcatggtg
gttggatatggtgttgatgaaggggatacgctaccacttgacctccctggtggcagtatt
gactgggttagtgagttcccctatcttgggtccctcatagcagagagtgggagggcacat
gaagaggtcgataggaggattgctagtgcgtccagagcatttggtgctctacgacagccc
atttttgaggaccctgacctgtccatcacaaccaagagatacatctacagagcctgtgtc
ttatcagtactgctgtatggaagtgagtgttgggtgccccttaagtga
>g536.t1
atggcgtcaggtaccatggttgatcaccactcgcaacttgcggccttatctgacgtttct
tttggggactgggtctctattgacactgaactccgctcacgactgtctcctatccattct
gatcttgctggtggtctcattgagattaccgaggcaggagatcttttttcatcgacagtc
tcctcgttgttgagcacgtgtcacccttcccctcgctccaccaaggcaccttcgggccct
caccgtcctagagcaactgaaaagctattaactaccctaactgagaagaagaacattgcc
aggagaagttttcgggctgaccctgagacgtttctgagtgcggtccgtgtccataataag
tgtgtcaaggcgtcccgtaagtcttcttttcacaggtcagctgtacgacaggagagggct
tttagatctaatccatgggaatactcaaaatcagtctgttgccccaagtcctctgttaaa
cccacatttaccagccaggagtgttttgattactttcagcactctttcagttcagattgt
ggtcctccgtatctgggactccctgattgggtggtcggtcatttccaatcaattgctgtt
```

g536.t1 in pep file.

```
>jg2229.t1
MEVEKSFGLTLSFEKIKFMVVGYGVDEGDTLPLDLPGGSIDWVSEFPYLGSLIAESGRAH
EEVDRRIASASRAFGALRQPIFEDPDLSITTKRYIYRACVLSVLLYGSECWVPLK*
>g536.t1
MASGTMVDHHSQLAALSDVSFGDWVSIDTELRSRLSPIHSDLAGGLIEITEAGDLFSSTV
SSLLSTCHPSPRSTKAPSGPHRPRATEKLLTTLTEKKNIARRSFRADPETFLSAVRVHNK
CVKASRKSSFHRSAVRQERAFRSNPWEYSKSVCCPKSSVKPTFTSQECFDYFQHSFSSDC
GPPYLGLPDWVVGHFQSIAVESPFDMSPITPAMVRGALRKCSNGSTPGPDKISYFHLKNL
PCTHHFLATLFSKILFVSHEAPLSWCMARITLIHKKGDSDSPSNFRPIALTSTVGKLFHR
IIARRLEHYLLVNSFVDSTIQKGFLSGIDGVIEHILSLNSIINNSRTNNLPLFLTFIDLR
NAFGSVHHSYIRDILSLIKLPPPVVSYIANLYSHLSASVCTQQWSTPSFKISRGVFQGDT
FSPVLFLLCFQPVIMNTVAQPSDGYRLKIPLQNTVGLPPIESYIYVEWHETDSSEPPGWY
HCQVIEYHVDSSATLVYRDGATESIDLHSVKWFFTRKNGKAYLPTDTTPPTYPLKKQRDA
AMLPKYAWSDSHKSKAYADDLTLLSVSKQDHQLALSSLSRSCQDLGFEIRPDKCVSYCFD
GHKSLPRLSFSLIEGHTTNISVGPTKFLGETLGITPTQSKRLSSKKLSSKVFEALERINE
RPIRGEYKVWIYKSYLIPSVSFNLTVDRISLATSKKIQTKITSYLKRWLKLPRCATLASL
FHPQVLNLPYLPHQLEKAKLRFLASTTLSSDHNVQALAAIVCDPQFIESEDIPRNSADLL
RSCQLPKKKSAMKKLYTALTDRHATEWNNHLESLSVQGKFLELIPLELESSVWSRILTSL
PAGQLSFLLRAGIDCLPTPVTLSRWHYQCNTSCKLCHSHSCTVHHVLNCCPASLNQGRYT
WRHDSVLKHLSSILKSNVSPDTVIFVDLPGFRAVDSPPSTIPPHILSTSARPDIVILEGN
VITLLELTIPINTKIGLQNARARKQAKDNYISLLGDLEARGFTSMLETVEIGSLGHFSLD
SINSLHSIISLLNRRSVRNLMLSLSRVAISCSQIIFNCRNQLEWSSPPLV*
>jg2349.t1
MTLSGFLACINGLSVFHIKVHNKYTADDFDDDLCTVLRRAGCCDEKIVFIMDESNVLDSG
FLERMNTLLAIGEGETGTLGTGLNSTMRRDRILKNSSFISALVSRRSERQWNSSVTGWCF
HSSKIVHCL*
```

g536.t1 sequence is from a soft masked region. Can it be an issue since genomes normally include softmasked regions?

**Comment by rhysf** [comment]
Hi @mudithekanayake 

Create_full_repo_sequence_databases.pl is by default pulling out all the mRNA features from your GFF.

From your grep output, your starting gff 'Ndig_braker.annotation.gff3' does not appear to have any mRNA feature for g536.t1, only for jg536.t1, which as you've said, is a different sequence:

contig_3519_np1212 AUGUSTUS mRNA 39886 41449 . + . ID=jg536.t1;Parent=jg536;

Looking through the code for Create_full_repo_sequence_databases.pl, the reason that it does not report an error, is because it is looping through every GFF entry and looking for corresponding FASTA entries - which is passing in your case. However, it was not doing the vice versa i.e. checking every FASTA entry and seeing if there is a corresponding GFF entry (which is not working in your case).  

Given this issue is likely to come up again, I have updated the code Create_full_repo_sequence_databases.pl to catch this issue. Additionally, I've added a verbose parameter to the script that will warn all id's that are not matching in both files.

I recommend re-downloading Synima with the new code, and trying to run the first step again on your input files in the way i describe above (using the verbose flag). When you have a full list of genes that are present in your FASTA files, but not in your GFF, you can then choose to update the GFF to include entries for them or omit them from your FASTA files. Once that is done, the pipeline should hopefully work for you.

Best,
Rhys

**Comment by mudithekanayake** [comment]
Hello @rhysf 

Thank you so much for your help. I will run it again using the updated version and remove the incompatible fasta sequences or update the gff file. As you think, will it give the same error if I use the --gene flag instead of the default mRNA since there are no gene features for g536.t1? (It does not show any "gene" feature when grep. Or doesn't it need gene features in --gene mode?).

**Comment by rhysf** [comment]
please do let me know if this solves the issue. Using the gene flag might also be suitable for you - i'd probably opt for whichever option captures the most number of features.

**Comment by mudithekanayake** [comment]
Hello @rhysf 
Finally it worked well up to the final step. Thank you very much for your kind support throughout the attempts. Since there are many contigs, my final output file looks like this.
[config.txt1.pdf](https://github.com/rhysf/Synima/files/8557330/config.txt1.pdf)
Contig names cover half of the figure. Do you have any suggestions to make it better. Also I would be grateful to you if you have any other suggestions to improve the figure by altering the parameters. (I changed the width and color but it still the same).
[config.txt5.pdf](https://github.com/rhysf/Synima/files/8557369/config.txt5.pdf)

**Comment by rhysf** [comment]
glad to hear and see it. It looks like your genomes are quite divergent from the plot. I'll therefore close this issue. 

But as for your new issue the simplest option is probably:

1) In the config file, you can choose words to remove from the id names e.g. 

contig_name_replace=Supercontig:sc,Contig:c,Scaffold:,Chromosome:chr

perhaps in your case, you would include scaffold e.g.

contig_name_replace=scaffold,whatever_else2,whatever_else3

2) more work would be re-naming all the contig names in your genome.fasta, coords and coords.spans.

Hope that helps. You might also want to play around with the size of the plot. And illustrator can help refine the figure further.

**Comment by mudithekanayake** [comment]
Hello @rhysf 
Thank you so much. Really appreciate it.
